pkg/addonmanager/controllers/agentdeploy/utils.go

```diff
@@ -128,34 +138,181 @@ func isPreDeleteHookObject(obj runtime.Object) (bool, *workapiv1.ManifestConfigO
 		},
 	}
 }
+func newManagedManifestWorkBuilder() *manifestWorkBuiler {
+	return &manifestWorkBuiler{
+		processor: &managedManifest{},
+	}
+}
+
+func newHostingManifestWorkBuilder() *manifestWorkBuiler {
+	return &manifestWorkBuiler{
+		processor: &hostingManifest{},
+	}
+}
+
+type manifestWorkBuiler struct {
+	processor manifestProcessor
+}
+
+type manifestProcessor interface {
+	deployable(installMode string, obj runtime.Object) (bool, error)
+	mutateManifest(installMode string, rawObject []byte) ([]byte, error)
+	manifestWorkName(addonNamespace, addonName string) string
+	preDeleteHookManifestWorkName(addonNamespace, addonName string) string
+}
+
+// hostingManifest process manifests which will be deployed on the hosting cluster
+type hostingManifest struct {
+}
+
+func (m *hostingManifest) deployable(installMode string, obj runtime.Object) (bool, error) {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return false, nil
+	}
+
+	location, exist, err := constants.GetHostedManifestLocation(accessor.GetLabels())
+	if err != nil {
+		return false, err
+	}
+	if installMode != constants.InstallModeHosted {
+		return false, nil
+	}
+
+	if exist && location == constants.HostedManifestLocationHosting {
+		klog.V(4).Infof("will deploy the manifest %s/%s on the hosting cluster in Hosted mode",
+			accessor.GetNamespace(), accessor.GetName())
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (m *hostingManifest) mutateManifest(installMode string, rawObject []byte) ([]byte, error) {
+	return rawObject, nil
```

Comment:
prupose?

Reply:
This method is currently not required and has been removed